File: app/src/main/java/com/getman/varnabeach/lifecycle/BeachConditionsViewModel.java
package com.getman.varnabeach.lifecycle;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.getman.varnabeach.data.Repository;

import java.util.Map;

public class BeachConditionsViewModel extends AndroidViewModel {

    private MutableLiveData<Map<String, String>> conditions;
    private final Repository repository;

    public BeachConditionsViewModel(Application application) {
        super(application);

        repository = Repository.getInstance(application);
    }

    public LiveData<Map<String, String>> getConditions(String lat, String lng, OnLoadingListener listener) {
        if (conditions == null) {
            conditions = new MutableLiveData<>();
            listener.onLoading();
            requestConditions(lat, lng);
        }
        return conditions;
    }

    private void requestConditions(String lat, String lng) {
        repository.requestNewConditions(lat, lng, this::updateData);
    }

    private void updateData() {
        conditions.setValue(repository.getBeachConditions());
    }
}

File: app/src/main/java/com/getman/varnabeach/lifecycle/BeachConditionsViewModel.kt
package com.getman.varnabeach.lifecycle

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.getman.varnabeach.data.Repository
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class BeachConditionsViewModel @Inject constructor(
    private val repository: Repository
) : ViewModel() {
    private var conditions: MutableLiveData<Map<String, String>> = MutableLiveData()
    val currentConditions get(): LiveData<Map<String, String>> = conditions

    fun requestConditions(
        lat: String?,
        lng: String?,
        listener: OnLoadingListener
    ) {
        if (lat != null && lng != null) {
            listener.onLoading()
            repository.requestNewConditions(lat, lng) { updateData() }
        }
    }

    private fun updateData() {
        conditions.value = repository.currentBeachConditions
    }
}